### PR TITLE
Create CCX view should send a JSON encoded payload

### LINKED ISSUE #134: The request to EDX to create a CCX should contain a JSON encoded payload
Bug found while writing the script to create CCX using CCXCon.

This is as easy as changing the request in `views.py` -> `create_ccx` from 

```
   try:
        resp = requests.post(
            '{instance}/api/ccx/v0/ccx/'.format(instance=course.edx_instance.instance_url),
            data=payload,
            headers={
                'Authorization': 'Bearer {}'.format(access_token),
            })
```

to

```
    try:
        resp = requests.post(
            '{instance}/api/ccx/v0/ccx/'.format(instance=course.edx_instance.instance_url),
            json=payload,
            headers={
                'Authorization': 'Bearer {}'.format(access_token),
            })
```

and update tests

### PR DESCRIPTION
**Pull request by giocalitri**
#### What are the relevant tickets?

fixes #134 
#### What's this PR do?

Fixes a bug in the creation of the CCX from CCXCon when a list of modules was passed
#### Where should the reviewer start?

`views.py`
#### How should this be manually tested?

Use the client script to create a CCX and pass some chapters modules.
#### Any background context you want to provide?

You need an EDX instance configured to work with CCXCon

### CODE REVIEW

File: courses/views_test.py [hunk 1]
```diff
@@ -486,6 +486,13 @@ def test_201_on_happy_case_with_modules(self):
                 content_type="application/json"
             )
         assert result.status_code == 201, result.content.decode('utf-8')
+        # check that the payload has been sent to edx as json
+        # this means that the requests library will encode it properly
+        # and add the `application/json` content type header
+        call_to_edx = mock_req.post.call_args
+        assert len(call_to_edx) == 2
```
Comment (noisecapella) [inline courses/views_test.py]: The return value from `call_args` is always 2 unless the mock wasn't called yet: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_args

File: courses/views_test.py [hunk 2]
```diff
@@ -486,6 +486,13 @@ def test_201_on_happy_case_with_modules(self):
...
+        call_content = call_to_edx[1]
+        assert 'json' in call_content
```
Comment (noisecapella) [inline courses/views_test.py:494]: Can you also assert the payload we're sending?

**Comment by giocalitri** [comment]
@noisecapella done

File: courses/views_test.py [hunk 3]
```diff
@@ -486,6 +486,25 @@ def test_201_on_happy_case_with_modules(self):
...
+        paylod_to_edx = {
+            'course_modules': [course_module.locator_id for course_module in course_modules],
+            'coach_email': payload['user_email'],
+            'master_course_id': course.course_id,
+            'display_name': payload['display_name'],
+            'max_students_allowed': payload['total_seats']
+        }
+        for key, value in paylod_to_edx.iteritems():
```
Comment (noisecapella) [inline courses/views_test.py]: I think you need to use `items` here instead of `iteritems` since Python 3 doesn't have `iteritems`

Comment (noisecapella) [inline courses/views_test.py]: Also can you rename `paylod_to_edx` to `payload_to_edx`?

File: courses/views_test.py [hunk 4]
```diff
@@ -486,6 +486,25 @@ def test_201_on_happy_case_with_modules(self):
...
+            assert key in call_content['json']
+            if not isinstance(value, list):
+                assert value == call_content['json'][key]
+            else:
+                assert sorted(value) == sorted(call_content['json'][key])
```
Comment (noisecapella) [inline courses/views_test.py:507]: Does the order we send the modules matter here? If not I'm fine with this

Comment (giocalitri) [inline courses/views_test.py:507]: no it does not, but the `assert` (instead of `self.assertEqual`) complains if they are not sorted in the same way.

**Comment by giocalitri** [comment]
@noisecapella done again

**Comment by noisecapella** [comment]
I think you missed my comment on renaming the variable :)

**Comment by giocalitri** [comment]
yeah, github had not updated the comment before I pushed. It should be fine now

**Comment by noisecapella** [comment]
:+1:

**Pull request merged**